ostd/src/orpc/framework/notifier.rs

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Arbitrary notifications via ORPC
+use alloc::string::ToString;
+
+use orpc_macros::orpc_trait;
+
+use crate::orpc::{
+    framework::errors::RPCError,
+    oqueue::{OQueueRef, ringbuffer::MPMCOQueue},
+};
+
+/// ORPC trait for a simple notifier
+#[orpc_trait]
+pub trait Notifier {
+    /// Notifies the server.
+    fn notify(&self) -> Result<(), RPCError>;
```

Comment:
Why have a method that forces a notification? It's a kind of odd feature for something that presumably is supposed to notify of some specific event.